test/system/libpod_suite_test.go

```diff
@@ -0,0 +1,696 @@
+	Distribution string
+	Version      string
+}
+
+// TestLibpod ginkgo master function
+func TestLibpod(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Libpod Suite")
+}
+
+var _ = BeforeSuite(func() {
+})
+
+// CreateTempDirin
+func CreateTempDirInTempDir() (string, error) {
+	return ioutil.TempDir("", "podman_test")
+}
+
+// Check if the key is seted in Env
+func envSeted(key string) bool {
+	_, set := os.LookupEnv(key)
+	return set
+}
+
+// PodmanCreate creates a PodmanTest instance for the tests
+func PodmanCreate(tempDir string) PodmanTest {
+	var envKey string
+	globalOptions := make(map[string]string)
+	podmanCmdOptions := make(map[string][]string)
+	//cwd, _ := os.Getwd()
+
+	for _, n := range GLOBALOPTIONS {
+		envKey = strings.Replace(strings.ToUpper(strings.Trim(n, "-")), "-", "_", -1)
+		if envSeted(envKey) {
+			globalOptions[n] = os.Getenv(envKey)
+		}
+	}
+
+	for _, n := range PODMAN_SUBCMD {
+		envKey = strings.Replace("PODMAN_SUBCMD_OPTIONS", "SUBCMD", strings.ToUpper(n), -1)
+		if envSeted(envKey) {
+			podmanCmdOptions[n] = strings.Split(os.Getenv(envKey), " ")
+		}
+	}
+
+	podmanBinary := "podman"
+	if os.Getenv("PODMAN_BINARY") != "" {
+		podmanBinary = os.Getenv("PODMAN_BINARY")
+	}
+
+	p := PodmanTest{
+		PodmanBinary:     podmanBinary,
+		ArtifactPath:     ARTIFACT_DIR,
+		TempDir:          tempDir,
+		GlobalOptions:    globalOptions,
+		PodmanCmdOptions: podmanCmdOptions,
+	}
+
+	// Setup registries.conf ENV variable
+	p.setDefaultRegistriesConfigEnv()
+	return p
+}
+
+//MakeOptions assembles all the podman main options
+func (p *PodmanTest) MakeOptions(args []string) []string {
+	var addOptions, subArgs []string
+	for _, n := range GLOBALOPTIONS {
+		if p.GlobalOptions[n] != "" {
+			addOptions = append(addOptions, n, p.GlobalOptions[n])
+		}
+	}
+
+	if len(args) == 0 {
+		return addOptions
+	}
+
+	subCmd := args[0]
+	addOptions = append(addOptions, subCmd)
+	if subCmd == "unmount" {
+		subCmd = "umount"
+	}
+	if subCmd == "help" {
+		subCmd = "h"
+	}
+
+	if _, ok := p.PodmanCmdOptions[subCmd]; ok {
+		m := make(map[string]bool)
+		subArgs = p.PodmanCmdOptions[subCmd]
+		for i := 0; i < len(subArgs); i++ {
+			m[subArgs[i]] = true
+		}
+		for i := 1; i < len(args); i++ {
+			if _, ok := m[args[i]]; !ok {
+				subArgs = append(subArgs, args[i])
+			}
+		}
+	} else {
+		subArgs = args[1:]
+	}
+
+	addOptions = append(addOptions, subArgs...)
+
+	return addOptions
+}
+
+// Podman is the exec call to podman on the filesystem, uid and gid the credentials to use
+func (p *PodmanTest) PodmanAsUser(args []string, uid, gid uint32, env []string) *PodmanSession {
+	podmanOptions := p.MakeOptions(args)
+	if os.Getenv("HOOK_OPTION") != "" {
+		podmanOptions = append(podmanOptions, os.Getenv("HOOK_OPTION"))
+	}
+	if env == nil {
+		fmt.Printf("Running: %s %s\n", p.PodmanBinary, strings.Join(podmanOptions, " "))
+	} else {
+		fmt.Printf("Running: (env: %v) %s %s\n", env, p.PodmanBinary, strings.Join(podmanOptions, " "))
+	}
```

Comment:
Not sure print env:nil will help anything.

Reply:
It helps from perspective of looking at the logs, reminder than command was called using the common/global environment.  Maybe this would be better:

```go
fmt.Printf("Running: (common env) %s %s\n", p.PodmanBinary, strings.Join(podmanOptions, " "))
```

It's not critical fix, just idea to help in debugging.